contracts/demo-issuer/src/lib.rs

```diff
@@ -78,14 +78,25 @@ impl Contract {
         ext_registry::ext(self.registry.clone())
             .with_attached_deposit(MINT_COST)
             .with_static_gas(Gas::ONE_TERA * 5) // 5 TGas
-            .sbt_mint(vec![(receiver, vec![metadata])]);
+            .sbt_mint(vec![(receiver, vec![metadata])])
+            .then(Self::ext(env::current_account_id()).sbt_mint_callback());
 
-        // TODO: add callback to undo identity insert if the minting didn't work because of a duplicate mint.
         if let Some(memo) = memo {
             env::log_str(&format!("SBT mint memo: {}", memo));
         }
     }
 
+    #[private]
+    pub fn sbt_mint_callback(
+        &mut self,
+        #[callback_result] last_result: Result<Vec<TokenId>, near_sdk::PromiseError>,
+    ) -> Option<Vec<TokenId>> {
```

Comment:
```suggestion
    ) -> Vec<TokenId> {
```

Reply:
Why returning option? It should be `Vec`. `last_result` is `Result<Vec<TokenId>, ...>` -- does this PR compile?